src/App.js

```diff
@@ -27,32 +27,58 @@ const store = createStore(
 );
 
 function App() {
-  const [token, setToken] = useState(localStorage.getItem("token"));
-  useEffect(() => reactGAinitialization(), []);
+  const [hasToken, setHasToken] = useState(localStorage.getItem("token"));
+  const [userid, setUserid] = useState();
+  useEffect(() => reactGAinitialization(), [userid]);
 
   return (
     <Provider store={store}>
       <Router>
         <div className="App" data-test={ifDev("App-component")}>
-            <Route
```

Comment:
This was moved towards the bottom. I was experimenting with the react-router-dom switch more after reading the docs on it. 

We are not using a switch statement because of the architecture of the application, and the navigation for components renders on the same path. The switch statement will only allow for the first component to be shown for that pathname.